runtime/codert_vm/thunkcrt.c

```diff
@@ -493,3 +558,56 @@ j9ThunkVMHelperFromNameAndSig(void * jitConfig, void *parm)
 
 	return j9ThunkVMHelperFromSignature(jitConfig, J9UTF8_LENGTH(J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSignature)), (char *) J9UTF8_DATA((J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSignature))));
 }
+
+const void *
+j9ThunkPersist(J9JITConfig *jitConfig, char *signatureChars, U_32 signatureLength, U_8 *thunkStart, U_32 totalSize)
+{
+	const void *store = NULL;
+
+#if defined(J9VM_OPT_SHARED_CLASSES)
+	J9JavaVM *vm = jitConfig->javaVM;
+	J9VMThread *vmThread = vm->internalVMFunctions->currentVMThread(vm);
+
+	U_8 encodedSignatureArray[2 * (J9_THUNK_MAX_ENCODED_BYTES - 1)];
+	U_8 *encodedSignature = encodedSignatureArray;
+	UDATA encodedLength;
+
+	J9SharedDataDescriptor dataDescriptor;
+	dataDescriptor.address = (U_8 *)thunkStart;
+	dataDescriptor.length = totalSize;
+	dataDescriptor.type = J9SHR_DATA_TYPE_AOTTHUNK;
+	dataDescriptor.flags = 0;
+
+	encodedLength = j9ThunkEncodeSignatureASCII(signatureChars, encodedSignature);
+
+	store = vm->sharedClassConfig->storeSharedData(vmThread, encodedSignature, encodedLength, &dataDescriptor);
+#endif
+
+	return store;
+}
+
+void *
+j9ThunkFindPersistentThunk(J9JITConfig *jitConfig, char *signatureChars, U_32 signatureLength, UDATA *thunkSize)
+{
+	void * thunk = NULL;
+
+#if defined(J9VM_OPT_SHARED_CLASSES)
+	J9JavaVM *vm = jitConfig->javaVM;
+	J9VMThread *vmThread = vm->internalVMFunctions->currentVMThread(vm);
+
+	U_8 encodedSignatureArray[2 * (J9_THUNK_MAX_ENCODED_BYTES - 1)];
+	U_8 *encodedSignature = encodedSignatureArray;
+	UDATA encodedLength;
+
+	J9SharedDataDescriptor dataDescriptor;
+	dataDescriptor.address = NULL;
+
+	encodedLength = j9ThunkEncodeSignatureASCII(signatureChars, encodedSignature);
+
+	vm->sharedClassConfig->findSharedData(vmThread, encodedSignature, encodedLength, J9SHR_DATA_TYPE_AOTTHUNK, FALSE, &dataDescriptor, NULL);
+	thunk = dataDescriptor.address;
+	*thunkSize = dataDescriptor.length;
+#endif
+
+	return thunk;
+}
```

Comment:
Pls add a newline at the end.

Reply:
Done in [force push](https://github.com/eclipse-openj9/openj9/compare/33381e4410338726a47c82fbe22e6c8200863274..227faa5f8dac0f99d13eae05b75a592d00346684).